### DISCUSSION TITLE
使用MVVM Toolkit引发的IntelliSense假错误如何解决

### DISCUSSION THREAD

**Discussion by Hakoyu** [BYJRK/DotNet-Discussions#17] [WPF 开发]
这些错误并不会影响软件运行(毕竟本来是正确的)
但如果出现了真的错误,就会和这些假错误混在一起,对debug造成阻碍
重启VS是可以解决问题,但过一会就又会出现了
有以下几种问题: 
- **未包含定义(实际上是有的)**
![image](https://user-images.githubusercontent.com/48956359/216759183-22b27a03-37b1-4110-a95b-51f0695d808f.png)
- **不存在名称(实际上是存在的)**
![image](https://user-images.githubusercontent.com/48956359/216759218-05abd75c-304d-4d2d-ab23-9899f4cc1e58.png)
- **二义性(姑且认为是和生成的文件冲突了)**
![image](https://user-images.githubusercontent.com/48956359/216759279-81d9b0ae-dcba-4bb2-803d-c3248d4aefe3.png)
![image](https://user-images.githubusercontent.com/48956359/216759400-e5efae97-8581-44f3-b20b-db880fb4b9ac.png)

**Comment by BYJRK** [comment]
奇怪，为什么你会有这种提示？我没遇到过。你的VS是什么版本？

**Reply by Hakoyu** [reply]
2022 我是一直保持最新版的

**Reply by BYJRK** [reply]
这也太邪乎了吧……你新建一个空项目，随便写一个ViewModel类试试，也是完全一样的效果吗？

**Reply by Hakoyu** [reply]
以前随便写了点是没遇到
现在写了好几个模型就开始出问题了
我清除一下工程文件试试

**Comment by TruePluto** [comment]
我觉得是频繁修改这些属性的代码导致的 我是用暴力的方法 删除`.vs`文件夹后重新打开解决

**Comment by Hakoyu** [comment]
我在 [CommunityToolkit](https://github.com/CommunityToolkit/dotnet/issues/598)看到了同样的问题
以及很久之前[Visual Studio Developer Community](https://developercommunity.visualstudio.com/t/IntelliSense-does-not-recognize-source-g/10242238?q=mvvm&sort=newest)就提到并解决了此类问题
不知道为什么又出现了
